pkg/store/bucket.go

```diff
@@ -1544,7 +1567,20 @@ func (r *bucketIndexReader) fetchPostings(groups []*postingGroup) error {
 				// Return postings and fill LRU cache.
 				// Truncate first 4 bytes which are length of posting.
 				groups[p.groupID].Fill(p.keyID, newBigEndianPostings(pBytes[4:]))
-				r.block.indexCache.StorePostings(r.ctx, r.block.meta.ULID, groups[p.groupID].keys[p.keyID], pBytes)
+
+				storeData := pBytes
+
+				if r.block.enablePostingsCompression {
+					// Reencode postings before storing to cache. If that fails, we store original bytes.
+					// This can only fail, if postings data was somehow corrupted,
+					// and there is nothing we can do about it. It's not worth reporting here.
+					data, err := diffVarintSnappyEncode(newBigEndianPostings(pBytes[4:]))
+					if err == nil {
```

Comment:
We will know as querying would fail?

Reply:
That was my thinking here... if it we get error here, it means postings was corrupted, so it will still be corrupted few lines later when it's used, and error will be reported there.

Reply:
These errors are now tracked via metric.